docs(user-guide): Add doc for import argocd packages (#11041) (#11096)
argoproj/argo-cd: docs/user-guide/import.md

> # Importing argocd packages 

Reviewer: ```suggestion
# Importing Argo CD go packages 
```

## Issue 

> When importing argocd packages in your own projects , you may face some erros when downloading the dependencies , such as unknown revision v0.0.0. This is because argocd directly depends on some kubernetes packages which have these unknown v0.0.0 version in their go.mod.

Reviewer: ```suggestion
When importing argocd packages in your own projects, you may face some errors when downloading the dependencies, such as "unknown revision v0.0.0". This is because argocd directly depends on some Kubernetes packages which have these unknown v0.0.0 version in their go.mod.
```

> When importing argocd packages in your own projects, you may face some errors when downloading the dependencies, such as "unknown revision v0.0.0". This is because argocd directly depends on some Kubernetes packages which have these unknown v0.0.0 version in their go.mod.

Reviewer: ```suggestion

When importing Argo CD packages in your own projects, you may face some errors when downloading the dependencies, such as "unknown revision v0.0.0". This is because Argo CD directly depends on some Kubernetes packages which have these unknown v0.0.0 versions in their go.mod.
```

## Solution

> The workaround is , you need to add a replace section in your own go.mod as same as the replace section of corresponding argocd version's go.mod. 

Reviewer: ```suggestion
Add a replace section in your own go.mod as same as the replace section of corresponding argocd version's go.mod. 
```

> Add a replace section in your own go.mod as same as the replace section of corresponding argocd version's go.mod. Inorder to find the go.mod for a specific version, navigate to the argocd repository and click on the switch branches/tags dropdown to select the version you are looking for. Now you can view the go.mod file for a specific version along with all other files.

Reviewer: ```suggestion

Add a replace section in your own go.mod as same as the replace section of the corresponding Argo CD version's go.mod. In order to find the go.mod for a specific version, navigate to the [Argo CD repository](https://github.com/argoproj/argo-cd/) and click on the switch branches/tags dropdown to select the version you are looking for. Now you can view the go.mod file for a specific version along with all other files.
```

## Example

> If you are using argocd v2.4.15, your go.mod should contain,

Reviewer: ```suggestion
If you are using Argo CD v2.4.15, your go.mod should contain the following:
```

# Importing Argo CD go packages 

## Issue 

When importing Argo CD packages in your own projects, you may face some errors when downloading the dependencies, such as "unknown revision v0.0.0". This is because Argo CD directly depends on some Kubernetes packages which have these unknown v0.0.0 versions in their go.mod.

## Solution

Add a replace section in your own go.mod as same as the replace section of the corresponding Argo CD version's go.mod. In order to find the go.mod for a specific version, navigate to the [Argo CD repository](https://github.com/argoproj/argo-cd/) and click on the switch branches/tags dropdown to select the version you are looking for. Now you can view the go.mod file for a specific version along with all other files.

## Example

If you are using Argo CD v2.4.15, your go.mod should contain the following:

> replace (

Reviewer: ```suggestion
```
replace (
```

# Importing Argo CD go packages 

## Issue 

When importing Argo CD packages in your own projects, you may face some errors when downloading the dependencies, such as "unknown revision v0.0.0". This is because Argo CD directly depends on some Kubernetes packages which have these unknown v0.0.0 versions in their go.mod.

## Solution

Add a replace section in your own go.mod as same as the replace section of the corresponding Argo CD version's go.mod. In order to find the go.mod for a specific version, navigate to the [Argo CD repository](https://github.com/argoproj/argo-cd/) and click on the switch branches/tags dropdown to select the version you are looking for. Now you can view the go.mod file for a specific version along with all other files.

## Example

If you are using Argo CD v2.4.15, your go.mod should contain the following:

> ```bash

Reviewer: ```suggestion
```
```
